cartographer_ros/cartographer_ros/occupancy_grid.cc

```diff
@@ -58,31 +60,34 @@ Eigen::AlignedBox2f ComputeMapBoundingBox(
 namespace cartographer_ros {
 
 void BuildOccupancyGrid2D(
-    const std::vector<::cartographer::mapping::TrajectoryNode>&
-        trajectory_nodes,
+    const std::vector<std::vector<::cartographer::mapping::TrajectoryNode>>&
+        all_trajectory_nodes,
     const string& map_frame,
     const ::cartographer::mapping_2d::proto::SubmapsOptions& submaps_options,
     ::nav_msgs::OccupancyGrid* const occupancy_grid) {
-  CHECK(!trajectory_nodes.empty());
+  CHECK(!all_trajectory_nodes.empty());
   namespace carto = ::cartographer;
   const carto::mapping_2d::MapLimits map_limits =
-      ComputeMapLimits(submaps_options.resolution(), trajectory_nodes);
+      ComputeMapLimits(submaps_options.resolution(), all_trajectory_nodes);
   carto::mapping_2d::ProbabilityGrid probability_grid(map_limits);
   carto::mapping_2d::RangeDataInserter range_data_inserter(
       submaps_options.range_data_inserter_options());
-  for (const auto& node : trajectory_nodes) {
-    if (node.trimmed()) {
-      continue;
+  carto::common::Time latest_time =
+      carto::common::FromUniversal(std::numeric_limits<int64_t>::min());
+  for (const auto& trajectory_nodes : all_trajectory_nodes) {
+    for (const auto& node : trajectory_nodes) {
+      if (node.trimmed()) {
+        continue;
+      }
+      latest_time = std::max(latest_time, node.time());
+      CHECK(node.constant_data->range_data_3d.returns.empty());  // No 3D yet.
```

Comment:
Should this be a `TODO` or should be remove the support for 3D from `ComputeMapBoundingBox()`?